backend/homework/2/src/main/java/org/q2/SentimentAnalyzer.java

```diff
@@ -0,0 +1,102 @@
+package org.q2;
+
+import java.util.*;
+import java.util.logging.Logger;
+public class SentimentAnalyzer {
+    static Logger logger= Logger.getLogger(String.valueOf(SentimentAnalyzer.class));
+    public static int[] detectProsAndCons(String review,String[][]featureSet,String[]posOpinionWords,String[]negOpinionWords)
+    {
+            int []featureOpinions=new int [featureSet.length];
+            for (int i=0;i<featureSet.length;i++)
+            {
+                int rating=0;
+                for(int j=0;j<featureSet[i].length;j++) {
+                    String feature=featureSet[i][j];
+                    int curRating = getOpinionOnFeature(review,feature,posOpinionWords,negOpinionWords);
+                    if(curRating!=0)
+                    {
+
+                        rating=curRating;
+                        break;
+                    }
+                }
+
+                featureOpinions[i]=rating;
+            }
+            return featureOpinions;
+    }
+    private static int getOpinionOnFeature(String review, String feature, String[] posOpinionWords, String[] negOpinionWords) {
+            int rating=checkForWasPhrasePattern(review,feature,posOpinionWords,negOpinionWords);
+            if(rating!=0)
+                return rating;
+            rating=checkForOpinionFirstPattern(review,feature,posOpinionWords,negOpinionWords);
+            return rating;
+    }
+    private static int checkForWasPhrasePattern(String review, String feature, String[] posOpinionWords, String[] negOpinionWords)
+    {
+        int opinion=0;
+        feature=feature+" was ";
+        for(int i=0;i<posOpinionWords.length;i++)
+        {
+            if(review.contains(feature.concat(posOpinionWords[i]))){
+                opinion=1;
+                return opinion;
+            }
+
+        }
+        for(int i=0;i< negOpinionWords.length;i++)
+        {
+            if(review.contains(feature.concat(negOpinionWords[i])))
+            {
+                opinion=-1;
+
+                return opinion;
+            }
+        }
+        return opinion;
```

Comment:
This and the above loop can be extracted in a single function made generic to fullfill both the needs